src/RFSCTraceBuilder.h

```diff
@@ -317,6 +282,30 @@ class RFSCTraceBuilder final : public TSOPSOTraceBuilder{
      * explored traces.
      */
     int sleep_branch_trace_count;
+
+    /* Pointer to the corresponing DecisionNode. */
+    std::shared_ptr<DecisionNode> decision_ptr;
+    
+    int get_decision_depth() const {
+      return decision_depth;
+    };
```

Comment:
It seems that it is always the case that `decision_depth == decision_ptr ? decision_ptr->depth : -1`
If so, why not remove this field and just return `decision_ptr ? decision_ptr->depth : -1`?